pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/bestpractices/LiteralsFirstInComparisonsRule.java

```diff
@@ -147,6 +159,25 @@ private boolean isStringLiteral(JavaNode node) {
         return false;
     }
 
+    private boolean isConstantString(JavaNode node) {
+        if (node instanceof ASTName) {
+            ASTName name = (ASTName) node;
+            ASTClassOrInterfaceBody classBody = name.getFirstParentOfType(ASTClassOrInterfaceBody.class);
+            ASTClassOrInterfaceBodyDeclaration classOrInterfaceBodyDeclaration = classBody.getFirstChildOfType(ASTClassOrInterfaceBodyDeclaration.class);
+            List<ASTFieldDeclaration> fieldDeclarations = classOrInterfaceBodyDeclaration.findChildrenOfType(ASTFieldDeclaration.class);
+            for (ASTFieldDeclaration fieldDeclaration : fieldDeclarations) {
+                ASTVariableDeclarator declaration = fieldDeclaration.getFirstChildOfType(ASTVariableDeclarator.class);
+                if (declaration.getName().equals(name.getImage())
+                        && "class java.lang.String".equals(declaration.getType().toString())
```

Comment:
This can be written as

```suggestion
                        && java.lang.String.class.equals(declaration.getType())
```